app/controllers/hyrax/generic_works_controller.rb

```diff
@@ -68,9 +81,17 @@ def download
 
   def assign_visibility
     if params["isDraft"] == Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
-     params["generic_work"]["visibility"] = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
+      params["generic_work"]["visibility"] = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
     else
       params["generic_work"]["visibility"] = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC
+      if params['action'] == 'update'
+        if params['id']
+          generic_work = GenericWork.find(params['id'])
+          if generic_work.visibility == Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
+            @updating_visibility_to_public = 'true'
```

Comment:
Why is the `true` a string? (not picking nits, just always sure I'm missing something)